sky/shell/gpu/rasterizer.cc

```diff
@@ -35,33 +36,35 @@ void Rasterizer::OnAcceleratedWidgetAvailable(gfx::AcceleratedWidget widget) {
   CHECK(surface_) << "GLSurface required.";
 }
 
-void Rasterizer::Draw(PassRefPtr<SkPicture> picture, gfx::Size size) {
+void Rasterizer::Draw(scoped_ptr<LayerTree> layer_tree) {
   TRACE_EVENT0("sky", "Rasterizer::Draw");
 
   if (!surface_)
     return;
 
+  gfx::Size size(layer_tree->frame_size().width(),
+                 layer_tree->frame_size().height());
+
+   if (layer_tree->root_layer())
```

Comment:
Yeah, this whole thing is messy.  I'll try again.  The problem is that I wanted to pass the size with the root layer because they're tied together, but I need to pass the size all the time, even when the dart code fails to produce a root layer.